### PR TITLE
8plots3

### PR DESCRIPTION
**Pull request by vladdez**

### CODE REVIEW

File: src/plot_erp.jl [hunk 1]
```diff
@@ -195,30 +196,45 @@ function plot_erp!(
...
+        mainAxis = Axis(
+            f_grid;
+            config.axis...,
+            xlabel = config.axis.xlabel,
+            ylabel = config.axis.ylabel,
+            yticklabelsize = config.axis.yticklabelsize,
+        )
         hidedecorations!(mainAxis, label = false, ticks = false, ticklabels = false)
-
+        hidespines!(mainAxis, :t, :r)
```
Comment (behinger) [inline src/plot_erp.jl:210]: this default should be in config.axis

File: src/plot_erp.jl [hunk 2]
```diff
@@ -195,30 +196,45 @@ function plot_erp!(
...
-
-        drawing = draw!(f_grid, plotEquation; axis = config.axis)
+        mainAxis = Axis(
+            f_grid;
+            config.axis...,
+            xlabel = config.axis.xlabel,
+            ylabel = config.axis.ylabel,
+            yticklabelsize = config.axis.yticklabelsize,
+        )
+        hidedecorations!(
```
Comment (behinger) [inline src/plot_erp.jl:225]: these defaults should be in config.axis

File: src/plot_erpgrid.jl [hunk 3]
```diff
@@ -29,25 +28,26 @@ function plot_erpgrid!(
...
-
     rel_zeropoint = argmin(abs.(times)) ./ length(times)
 
     for (ix, p) in enumerate(eachcol(pos))
         x = p[1] #- 0.1
         y = p[2] #- 0.1
         # todo: 0.1 should go into plot config
         ax = Axis(
-            f[1, 1],
+            f[1:8, 1:8],
```
Comment (behinger) [inline src/plot_erpgrid.jl:50]: I dont understand why this is necessary - 1:8 seems arbitrary?

File: src/plot_erpgrid.jl [hunk 4]
```diff
@@ -82,6 +82,26 @@ function plot_erpgrid!(
     hidedecorations!.(axlist)
     hidespines!.(axlist)
 
+    ax2 = Axis(f[1:8, 1:8]) #, xlabel = config.axis.xlabel, ylabel = config.axis.ylabel)
```
Comment (behinger) [inline src/plot_erpgrid.jl:85]: is this for the label inset?

Why not use the same relative position as for the other axes?

Is there any check that it doesnt overlay with an actual axis?

File: src/plot_erpimage.jl [hunk 5]
```diff
@@ -13,7 +10,7 @@ Plot an ERP image.
 ## Keyword Arguments
 - `erpblur` (Number, `10`): number indicating how much blur is applied to the image; using Gaussian blur of the ImageFiltering module.
 Non-Positive values deactivate the blur.
-- `sortix` (default: `nothing`): .
+- `sortix` (default: `nothing`): ???.
```
Comment (behinger) [inline src/plot_erpimage.jl:13]: sortix allows to specify the ordering manually.

File: src/plot_erpimage.jl [hunk 6]
```diff
@@ -13,7 +10,7 @@ Plot an ERP image.
...
 - `sortvalues` (bool, `false`): parameter over which plot will be sorted. Using sortperm() of Base Julia.
```
Comment (behinger) [inline src/plot_erpimage.jl:14]: sortvalues is not bool, but rather a vector to be sorted. - not sure what the default is

File: src/plotconfig.jl [hunk 7]
```diff
@@ -149,7 +151,7 @@ function PlotConfig(T::Val{:topoplotseries})
     config_kwargs!(
         cfg,
         layout = (use_colorbar = true,),
-        colorbar = (; flipaxis = true, labelrotation = -π / 2, label = "Voltage [µV]"),
+        colorbar = (; flipaxis = true, labelrotation = -π / 2, label = "Voltage [µV]"), # is it a duplication??
```
Comment (behinger) [inline src/plotconfig.jl:154]: duplication of what?

**Review by behinger — COMMENTED** [review]
lot's of changes. Would be great next time to split it up in multiple PRs, then we could merge some already and other who need more work can stay.

But that makes it more painful to work there ;-)

File: src/layout_helper.jl [hunk 8]
```diff
@@ -9,16 +15,22 @@ function applyLayoutSettings!(config::PlotConfig; fig=nothing, hm=nothing,
...
+                            legend_position;
+                            colormap = config.visual.colormap,
+                            config.colorbar...,
+                        )
                     else
                         Colorbar(legend_position, hm; config.colorbar...)
                     end
                 else
-                    Legend(legend_position, ax; config.legend...)
+                    Legend(legend_position, ax, "Conditions"; config.legend...)
```
[the remark this answers is not in the document]
Comment (vladdez) [reply src/layout_helper.jl:33]: yes, how I put config before semicolon? and the name should be before it

File: docs/example_data.jl [hunk 9]
```diff
@@ -90,19 +90,18 @@ function example_data(example = "TopoPlots.jl")
...
-    elseif example == "sort_data"
+    elseif example == "sort_data" #this should be reviewed
         dat, evts =
             UnfoldSim.predef_eeg(; onset = LogNormalOnset(μ = 3.5, σ = 0.4), noiselevel = 5)
         dat_e, times = Unfold.epoch(dat, evts, [-0.1, 1], 100)
         evts, dat_e = Unfold.dropMissingEpochs(evts, dat_e)
-        evts.Δlatency = diff(vcat(evts.latency, 0)) *-1
+        evts.Δlatency = diff(vcat(evts.latency, 0))
         dat_e = dat_e[1, :, :]
+        evts = filter(row -> row.Δlatency > 0, evts)
```
[the remark this answers is not in the document]
Comment (vladdez) [reply docs/example_data.jl:104]: there was an outlier

File: src/plot_erpgrid.jl (hunk 3 above)
Comment (vladdez) [reply src/plot_erpgrid.jl:50]: trays of old solution, I will remove it

**Comment by vladdez** [comment]
Thanks for comments!
Yes, I agree a lot of changes, they were all related with 8Plots

File: src/layout_helper.jl (hunk 8 above)
[the remark this answers is not in the document]
Comment (behinger) [reply src/layout_helper.jl:33]: maybe
`Legend(legend_position,ax, config.legend.title; config.legend ...)`

and legend.title = "" by default in config?

File: docs/example_data.jl (hunk 9 above)
[the remark this answers is not in the document]
Comment (behinger) [reply docs/example_data.jl:104]: ok, maybe check with my new proposal above, the outlier should be gone

File: docs/example_data.jl [hunk 10]
```diff
@@ -90,19 +90,18 @@ function example_data(example = "TopoPlots.jl")
...
         df.time = range(-0.3, 0.5, step = 1 / 500)[Int.(df.time)]
         return df, chanlocs
-    elseif example == "sort_data"
+    elseif example == "sort_data" #this should be reviewed
         dat, evts =
             UnfoldSim.predef_eeg(; onset = LogNormalOnset(μ = 3.5, σ = 0.4), noiselevel = 5)
         dat_e, times = Unfold.epoch(dat, evts, [-0.1, 1], 100)
         evts, dat_e = Unfold.dropMissingEpochs(evts, dat_e)
-        evts.Δlatency = diff(vcat(evts.latency, 0)) *-1
+        evts.Δlatency = diff(vcat(evts.latency, 0))
```
[the remark this answers is not in the document]
Comment (vladdez) [reply docs/example_data.jl:102]: originally it was without *-1
I used it as a workaround

File: src/layout_helper.jl (hunk 8 above)
[the remark this answers is not in the document]
Comment (vladdez) [reply src/layout_helper.jl:33]: didn't work

[the remark this answers is not in the document]
Comment (behinger) [reply src/layout_helper.jl:33]: why not?

File: src/plot_erpgrid.jl (hunk 4 above)
Comment (vladdez) [reply src/plot_erpgrid.jl:85]: done

File: src/plot_erpimage.jl (hunk 5 above)
Comment (vladdez) [reply src/plot_erpimage.jl:13]: What do you mean by manually? How it differs form sortvalues?

Comment (behinger) [reply src/plot_erpimage.jl:13]: sortix = [3,4,1,5,2]
sortval = [110 120 50 200 30]

Comment (vladdez) [reply src/plot_erpimage.jl:13]: I would rather rename it then, because current name is kinda meaningless

Comment (behinger) [reply src/plot_erpimage.jl:13]: sure, sortindex and sortvalues is better anyway - I think sortindex is a good name though

File: src/layout_helper.jl (hunk 8 above)
[the remark this answers is not in the document]
Comment (vladdez) [reply src/layout_helper.jl:33]: ERROR: type NamedTuple has no field title

[the remark this answers is not in the document]
Comment (behinger) [reply src/layout_helper.jl:33]: did you add it to the config?

[the remark this answers is not in the document]
Comment (vladdez) [reply src/layout_helper.jl]: of course

[the remark this answers is not in the document]
Comment (behinger) [reply src/layout_helper.jl]: so if you run `cfg = PlotConfig()` you can do `cfg.legend.title` without an error?

**Review by behinger — APPROVED** [review]

**Comment by behinger** [comment]
giphy

ready to merge!

**Pull request merged**